UM/Qt/qml/UM/CheckBox.qml

```diff
@@ -74,9 +76,10 @@ CheckBox
                 {
                     name: "partiallyChecked"
                     when: (control.checkState == Qt.PartiallyChecked)
-                    PropertyChanges {
+                    PropertyChanges
+                    {
                         target: checkIcon
-                        height: UM.Theme.getSize("checkbox_square").height
+                        height: parent.height - Math.round((parent.height - UM.Theme.getSize("checkbox_square").height) / 2) * 2 //Round border to pixels to properly center it.
```

Comment:
The checkbox still appears 1 pixel misaligned on the vertical for me. I don't think this is noticeable though.

![Screenshot 2022-02-01 at 17 23 10](https://user-images.githubusercontent.com/23387864/152126075-6971c8b4-9abf-4db6-87c3-fd55962b6928.png)

Reply:
I'm viewing this at a resolution of 1920x1080. System em size is 14px. On Ubuntu 20.04

Reply:
This is what I'm seeing:
![image](https://user-images.githubusercontent.com/2448634/152131322-828edc64-4495-49fb-bbd4-3e45bfcdc9d2.png)

Your picture is how it looked before my last commit in this PR! Something weird going on then.